diff --git a/config.sub b/config.sub
index 78176a4..1e8fa8f 100755
--- a/config.sub
+++ b/config.sub
@@ -1333,7 +1333,7 @@ case $os in
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* \
 	      | -amigaos* | -amigados* | -msdos* | -newsos* | -unicos* | -aof* \
-	      | -aos* | -aros* \
+	      | -aos* | -aros* | -exscapeos* \
 	      | -nindy* | -vxsim* | -vxworks* | -ebmon* | -hms* | -mvs* \
 	      | -clix* | -riscos* | -uniplus* | -iris* | -rtu* | -xenix* \
 	      | -hiux* | -386bsd* | -knetbsd* | -mirbsd* | -netbsd* \
diff --git a/newlib/configure.host b/newlib/configure.host
index ef73fd0..e31e784 100644
--- a/newlib/configure.host
+++ b/newlib/configure.host
@@ -406,6 +406,11 @@ case "${host}" in
   h8500-*-elf*)
 	sys_dir=h8500hms
 	;;
+  i[3-7]86-*-exscapeos*)
+	sys_dir=exscapeos
+	newlib_cflags="${newlib_cflags} -DMISSING_SYSCALL_NAMES -D__DYNAMIC_REENT__ -O0 -gstabs+" # -DREENTRANT_SYSCALLS_PROVIDED
+	posix_dir=posix
+	;;
   i[34567]86-*-rdos*)
 	sys_dir=rdos
 	newlib_cflags="${newlib_cflags} -DMISSING_SYSCALL_NAMES"
diff --git a/newlib/libc/sys/configure b/newlib/libc/sys/configure
index f1635da..d0f03ba 100755
--- a/newlib/libc/sys/configure
+++ b/newlib/libc/sys/configure
@@ -789,6 +789,7 @@ arc
 arm
 d10v
 decstation
+exscapeos
 h8300hms
 h8500hms
 linux
@@ -11435,7 +11436,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 11438 "configure"
+#line 11439 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -11541,7 +11542,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 11544 "configure"
+#line 11545 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -11794,6 +11795,8 @@ subdirs="$subdirs a29khif"
  ;;
 	decstation) subdirs="$subdirs decstation"
  ;;
+	exscapeos) subdirs="$subdirs exscapeos"
+ ;;
 	h8300hms) subdirs="$subdirs h8300hms"
  ;;
 	h8500hms) subdirs="$subdirs h8500hms"
diff --git a/newlib/libc/sys/configure.in b/newlib/libc/sys/configure.in
index 5a2b121..acbcf53 100644
--- a/newlib/libc/sys/configure.in
+++ b/newlib/libc/sys/configure.in
@@ -27,6 +27,7 @@ if test -n "${sys_dir}"; then
 	arm) AC_CONFIG_SUBDIRS(arm) ;;
 	d10v) AC_CONFIG_SUBDIRS(d10v) ;;
 	decstation) AC_CONFIG_SUBDIRS(decstation) ;;
+	exscapeos) AC_CONFIG_SUBDIRS(exscapeos) ;;
 	h8300hms) AC_CONFIG_SUBDIRS(h8300hms) ;;
 	h8500hms) AC_CONFIG_SUBDIRS(h8500hms) ;;
 	linux) AC_CONFIG_SUBDIRS(linux) ;;
